client/app/redux/actions/MenuItemInformationActions.js

```diff
@@ -26,7 +26,11 @@ export function getMenuItemInformation(itemID) {
     return async (dispatch) => {
         dispatch(request());
         try {
-            const menuItem = await get("/api/menuItems", { menuitemid: itemID });
+            const { nutrition, allergens, ingredients } = item;
+            const menuItem =
+                (nutrition && allergens && ingredients)
+                    ? item
+                    : await get("/api/menuItems", { menuitemid: item.itemID })
```

Comment:
Looks good to me. Were you able to test the cases where the information hasn't been cached?